internal/statemachine/snap_states.go

```diff
@@ -42,3 +44,29 @@ func (stateMachine *StateMachine) prepareImage() error {
 
 	return nil
 }
+
+// Generate the manifest
+func (stateMachine *StateMachine) generateSnapManifest() error {
+	// We could use snapd's seed.Open() to generate the manifest here, but
+	// actually it doesn't make things much easier than doing it manually -
+	// like we did in the past. So let's just go with this.
+
+	// snaps.manifest
+	outputPath := filepath.Join(stateMachine.commonFlags.OutputDir, "snaps.manifest")
+	snapsDir := filepath.Join(stateMachine.tempDirs.rootfs, "system-data", "var", "lib", "snapd", "snaps")
+	err := helper.WriteSnapManifest(snapsDir, outputPath)
+	if err != nil {
+		return err
```

Comment:
Ah here it is, this is the other line that isn't covered by tests. The helper.Write function could also be mocked.